ksqldb-execution/src/main/java/io/confluent/ksql/execution/expression/tree/LambdaLiteral.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.expression.tree;
+
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.parser.NodeLocation;
+
+import java.util.Objects;
+import java.util.Optional;
+
+@Immutable
+public class LambdaLiteral extends Literal {
```

Comment:
question rather than suggestion - why do we need to use another type for these? An alternative would be to just use `UnqualifiedColumnReferenceExp` (and maybe rename it to `UnqualifiedReferenceExp`). Do you think it makes it easier to implement lambdas if they have different types?

Reply:
I think when implementing the type reference, it makes the code cleaner. I'd imagine when doing type checking in `ExpressionTypeManager`, instead of having to pass down in the context (similar to what's happening in AstSanitizer) that we're in a lambda function expression, we can just have a specific visitor for a LambdaLiteral.

Reply:
After working through some of the type stuff, I do think it's somewhat necessary to have the additional type. Right now, the array/map input is passed in as an `UnqualifiedReferenceExp` and it's nice to be able to distinguish that from the lambda inputs